internal/app/kdk/pull.go

```diff
@@ -21,11 +21,16 @@ import (
 
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/client"
+	"io/ioutil"
 )
 
 func Pull(ctx context.Context, dockerClient *client.Client, imageCoordinates string) error {
 	out, err := dockerClient.ImagePull(ctx, imageCoordinates, types.ImagePullOptions{})
 	defer out.Close()
-	io.Copy(os.Stdout, out)
+	if Verbose {
+		io.Copy(os.Stdout, out)
+	} else {
+		io.Copy(ioutil.Discard, out)
```

Comment:
I don't understand why you might do this.  Maybe this is something I should know.  My understanding is that you are discarding data from a connection that is going to be closed by the defer statement above anyway.  Is this useful in some way?

Reply:
https://forum.golangbridge.org/t/do-i-need-to-read-the-body-before-close-it/5594

Reply:
Still interested:
https://github.com/google/go-github/pull/317

Reply:
@drekle Apparently, without the ioutil.Discard, the code would finish before the command causing the defer statement to trigger and kill the command early.  We specifically debugged this mystery.  If you're interested, comment out the io.Copy, and try running it. The pull will fail every time.